### PR TITLE
Fix zip compression

### LINKED ISSUE #1138: Unable to unzip 4.0-alpha1
I just downloaded versionpress-4.0-alpha1 and am getting an error when unzipping.  Have tried it several different ways.

screenshot 2016-10-10 15 03 13

### PR DESCRIPTION
**Pull request by Ishirak**
Resolves #1138.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -370,4 +371,52 @@ gulp.task('idea', 'Setup project files for IDEA / PhpStorm', function () {
 
 function isRelative(sitePath) {
     return path.normalize(sitePath) != path.resolve(sitePath);
+}
+
+/**
+ * Compress the build into a zip file.
```
Comment (borekb) [inline gulpfile.js]: There should be an explanation here why we're not using the `gulp-zip` package – a link to some issue or something.

**Comment by borekb** [comment]
@Ishirak, can you link to some `gulp-zip` issue or something similar which would explain what was the problem? Did you encounter anything useful during your research?

**Comment by Ishirak** [comment]
@borekb in the process of decompressing the zip file on the Mac I've encountered following error couple of times:

> symlink error: File name too long

In the process of fixing this error I've tried to update all of the packages in the `package.json`. No succes.

Next I've tried to compress the build using default Mac tools. Succes – problem had to be in the `gulp-zip` package.

So I've tried to use some alternatives to the `gulp-zip`, e.g. `gulp-gzip`, `gulp-tar` or `archiver`. In every single one I've tried I've encountered another issue, e.g. no support for single directory compression, compression of single files etc.

Only [gulp-tar](https://github.com/sindresorhus/gulp-tar) was almost perfect – written by the same author as [gulp-zip](https://github.com/sindresorhus/gulp-zip), but used different compression algorithm. Its only downside was the `.tar` suffix. So I've used `gulp-tar` as the source for the `compress()` function and changed its behaviour to support our use case – `.zip` suffix.

**Comment by borekb** [comment]
Wow, just to generate a stupid ZIP. 👍

**Comment by vasek17** [comment]
It is a bug in `gulp-zip` package, I reproduced and extracted the minimal example.
Related issue: https://github.com/sindresorhus/gulp-zip/issues/79.

**Pull request merged**